src/dockerconfig/docker_config.go

```diff
@@ -12,82 +12,120 @@ import (
 	"sigs.k8s.io/controller-runtime/pkg/client"
 )
 
-type DockerConfig struct {
-	ApiReader        client.Reader
-	Dynakube         *dynatracev1beta1.DynaKube
-	Auths            map[string]DockerAuth
-	TrustedCertsPath string
-}
+const (
+	TmpPath         = "/tmp/dynatrace-operator"
+	CADir           = "ca"
+	RegistryAuthDir = "dockerconf"
+)
 
-type DockerAuth struct {
-	Username string `json:"username"`
-	Password string `json:"password"`
+type DockerConfig struct {
+	ApiReader          client.Reader
+	Dynakube           *dynatracev1beta1.DynaKube
+	RegistryAuthPath   string
+	TrustedCertsPath   string
+	registryAuthSecret *corev1.Secret
 }
 
 func NewDockerConfig(apiReader client.Reader, dynakube dynatracev1beta1.DynaKube) *DockerConfig {
+	trustedCertsPath := ""
+	if dynakube.Spec.TrustedCAs != "" {
+		trustedCertsPath = path.Join(TmpPath, CADir, dynakube.Name)
+	}
 	dockerConfig := DockerConfig{
-		ApiReader: apiReader,
-		Auths:     make(map[string]DockerAuth),
-		Dynakube:  &dynakube,
+		ApiReader:        apiReader,
+		Dynakube:         &dynakube,
+		RegistryAuthPath: path.Join(TmpPath, RegistryAuthDir, dynakube.Name),
+		TrustedCertsPath: trustedCertsPath,
 	}
 	return &dockerConfig
 }
 
-func (config *DockerConfig) SetupAuths(ctx context.Context) error {
-	var pullSecret corev1.Secret
-	err := config.ApiReader.Get(ctx, client.ObjectKey{Name: config.Dynakube.PullSecret(), Namespace: config.Dynakube.Namespace}, &pullSecret)
+func (config *DockerConfig) StoreRequiredFiles(ctx context.Context, fs afero.Afero) error {
+	registryCredentials, err := config.getRegistryCredentials(ctx)
 	if err != nil {
-		log.Info("failed to load pull secret", "dynakube", config.Dynakube.Name)
-		return errors.WithStack(err)
+		return err
 	}
-	dockerAuths, err := parseDockerAuthsFromSecret(&pullSecret)
-	if err != nil {
-		log.Info("failed to parse pull secret content", "dynakube", config.Dynakube.Name)
+
+	if err := saveFile(registryCredentials, fs, config.RegistryAuthPath); err != nil {
+		log.Info("failed to store registry credentials", "dynakube", config.Dynakube.Name)
 		return err
 	}
-	config.Auths = dockerAuths
+
+	if config.Dynakube.Spec.TrustedCAs != "" {
+		config.TrustedCertsPath = path.Join(TmpPath, CADir, config.Dynakube.Name)
```

Comment:
a new function for the whole trustedCA part would make it way cleaner

Reply:
Created 2 new functions for trustedCAs and auths to make it clear what the code does